app/models/spree/credit_card_decorator.rb

```diff
@@ -15,7 +15,7 @@
   belongs_to :user
 
   after_create :ensure_single_default_card
-  after_save :ensure_single_default_card, if: :is_default_changed?
+  after_save :ensure_single_default_card, if: :default_card_needs_updating?
```

Comment:
```suggestion
  after_save :ensure_single_default_card
  
  def ensure_single_default_card
      return unless default_card_needs_updating?
```

I think these inline if statements are never a good idea and should be moved into the method itself.
actually, I think callbacks are never a good idea, in this case, that means that saving a credit card should be in a service.

Reply:
> I think these inline if statements are never a good idea and should be moved into the method itself.

I agree. The difficulty here is that we have two different events calling the same method (create and save events). They don't share the same conditions.

I also agree with avoiding callbacks. I opened a new tech debt issue: #3966. Please let me know what you think about the two modelling options.

Reply:
I see :+1:
I left my vote in the tech debt issue.